app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -195,7 +199,13 @@ object FormTemplatesControllerRequestHandler {
         case None => JsSuccess(())
       }
 
-    sectionValidations andKeep dmsFormIdValidations andKeep jsonValue.transform(
+    val declarationSectionValidations: JsResult[Unit] = (jsonValue \ "declarationSection" \ "fields").toOption match {
+      case Some(fields) if fields.as[JsArray].value.flatMap(jv => (jv \ "type").asOpt[String]).contains("group") =>
+        invalidGroupComponentInDeclarationSection
+      case _ => JsSuccess(())
+    }
+
+    declarationSectionValidations andKeep sectionValidations andKeep dmsFormIdValidations andKeep jsonValue.transform(
```

Comment:
We should not validate this on json level. Json level validation are supposed to be really low level.

We should check this on `FormTemplate` level. Best place is to add new check into [Verifier.scala](https://github.com/hmrc/gform/blob/master/app/uk/gov/hmrc/gform/formtemplate/Verifier.scala)